applications/asset_tracker/src/main.c

```diff
@@ -376,39 +381,108 @@ static void button_send(bool pressed)
 }
 #endif
 
+#if IS_ENABLED(CONFIG_GPS_START_ON_MOTION)
+static bool motion_activity_is_active(void)
+{
+	return (last_activity_state == MOTION_ACTIVITY_ACTIVE);
+}
+
+static void motion_trigger_gps(motion_data_t  motion_data)
+{
+	if (!gps_control_is_enabled()) {
+		return;
+	}
+
+	if (motion_activity_is_active()) {
+		static s64_t next_active_time;
+		s64_t last_active_time =
+			gps_control_get_last_active_time() / 1000;
+		s64_t now = k_uptime_get() / 1000;
+		s64_t time_since_fix_attempt = now - last_active_time;
+		s64_t time_until_next_attempt = next_active_time - now;
+
+		LOG_DBG("Last at %lld s, now %lld s, next %lld s; "
+			"%lld secs since last, %lld secs until next",
+			last_active_time, now, next_active_time,
+			time_since_fix_attempt, time_until_next_attempt);
+
+		if (time_until_next_attempt >= 0) {
+			LOG_DBG("keeping original schedule.");
+			return;
+		}
+
+		time_since_fix_attempt = MAX(0, (MIN(time_since_fix_attempt,
+			CONFIG_GPS_CONTROL_FIX_CHECK_INTERVAL)));
+		s64_t time_to_start_next_fix = 1 +
+			CONFIG_GPS_CONTROL_FIX_CHECK_INTERVAL -
+			time_since_fix_attempt;
+
+		next_active_time = now + time_to_start_next_fix;
+
+		char buf[100];
+
+		snprintf(buf, sizeof(buf),
+			"Motion triggering GPS; accel, %.1f, %.1f, %.1f",
+			motion_data.acceleration.x,
+			motion_data.acceleration.y,
+			motion_data.acceleration.z);
+		LOG_INF("%s", buf);
```

Comment:
Why not just LOG_INF("Motion triggering GPS; accel, %.1f, %.1f, %.1f". [...]); ?

Reply:
Agreed, especially that current approach would require to use `log_strdup` on the string buffer.

Reply:
I'll investigate more deeply why this was necessary and reply again.  But the reason at the time was that doing as you say resulted in only the integer portions of the values being printed; I assumed that meant that for logging, floating point support was disabled.  Rather than require a change to the configuration of the logging subsystem simply to print these values, I opted to snprintf to a local buffer, which worked.  

This logging occurs at most once every CONFIG_GPS_CONTROL_FIX_CHECK_INTERVAL seconds, which is currently set to 30.  Any additional computational load of executing log_strdup() is in the noise compared to powering up and obtaining a GPS fix.

Reply:
Ok, it sounds reasonable to use `snprintf` then, still you need to use `log_strdup`, otherwise it might not work in deferred logger mode, something like:
```
LOG_INF("%s", log_strdup(buf));
```

Reply:
Let's put this context in a comment in the code.

Reply:
(misplaced github comment moved here):
This is a known bug in Zephyr:
zephyrproject-rtos/zephyr#18351
zephyrproject-rtos/zephyr#18921

The solution seems to be what I actually did myself.

I'll add log_strdup().

Reply:
Thank you